test/integration/ControllerCurveUniV3Strat-optimism-test.js

```diff
@@ -0,0 +1,211 @@
+
+    [strat, wNativeFeed, crvFeed, opFeed] = await Promise.all([
+      ethers.getContractAt('ControllerCurveUniV3Strat', (await controller.strategy())),
+      ethers.getContractAt('IChainLink', '0x13e3Ee699D1909E989722E753853AE30b17e08c5'),
+      ethers.getContractAt('IChainLink', '0xbD92C6c284271c227a1e0bF1786F468b539f51D9'),
+      ethers.getContractAt('IChainLink', '0x0D276FC14719f9292D5C1eA2198673d1f4269246')
+    ])
+
+    await strat.setUseNative(true)
+
+    poolSlipage = 0.015
+
+    await Promise.all([
+      setChainlinkRoundForNow(wNativeFeed),
+      setChainlinkRoundForNow(crvFeed),
+      setChainlinkRoundForNow(opFeed),
+      waitFor(strat.setMaxPriceOffset(86400)),
+      waitFor(strat.setPriceFeed(WETH.address, wNativeFeed.address)),
+      waitFor(strat.setPriceFeed(CRV.address, crvFeed.address)),
+      waitFor(strat.setPriceFeed(OP.address, opFeed.address)),
+      waitFor(strat.setPoolSlippageRatio(poolSlipage * 10000)),
+      waitFor(strat.setSwapSlippageRatio(500)),
+      waitFor(strat.setRewardToWantRoute(OP.address, [OP.address, WETH.address])),
+      waitFor(strat.setRewardToWantRoute(CRV.address, [CRV.address, WETH.address])),
+      waitFor(strat.setTokenToTokenSwapFee(OP.address, WETH.address, 3000)),
+      waitFor(strat.setTokenToTokenSwapFee(CRV.address, WETH.address, 10000)),
+    ])
+  })
+
+  itIf(hre.network.config.network_id === 10, 'Full deposit + harvest strat + withdraw', async () => {
+    const currentBalance = async () => {
+      return await ethers.provider.getBalance(bob.address)
+    }
+
+    const initialBalance = await currentBalance()
+
+    await setWethBalanceFor(bob.address, '100', 3)
+    expect(await WETH.balanceOf(strat.address)).to.be.equal(0)
+    expect(await CurveRewardsGauge.balanceOf(strat.address)).to.be.equal(0)
+
+    await waitFor(WETH.connect(bob).approve(archimedes.address, '' + 100e18))
+    await waitFor(archimedes.connect(bob).deposit(0, '' + 100e18, zeroAddress))
+
+    expect(await WETH.balanceOf(controller.address)).to.be.equal(0)
+    expect(await WETH.balanceOf(strat.address)).to.be.equal(0)
+    expect(await CurveRewardsGauge.balanceOf(strat.address)).to.be.within(
+      99.2e18 + '', // production virtual price is ~1.003.
+      100e18 + ''
+    )
+
+    const balance = await strat.balanceOfPool() // more decimals
+
+    await mineNTimes(100)
+    expect(await strat.harvest()).to.emit(strat, 'Harvested')
+
+    expect(await strat.balanceOfPool()).to.be.above(balance)
+
+    // withdraw 95 WETH in shares
+    const toWithdraw = (
+      (await controller.totalSupply()).mul(95e18 + '').div(
+        await controller.balance()
+      )
+    )
+
+    await waitFor(archimedes.connect(bob).withdraw(0, toWithdraw))
+
+    expect((await currentBalance()).sub(initialBalance)).to.within(
+      94.9e18 + '',
+      95e18 + '' // 95 - 0.1% withdrawFee
+    )
+    expect(await WETH.balanceOf(strat.address)).to.equal(0)
```

Comment:
I would add the `expect(await ethers.provider.getBalance(strat.address)).to.be.equal(0)` too, to be sure that we don't lose anything